### PR TITLE
feat: Don't send TP approve meta transaction if merkle root is the same

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR updates the TP approval flow so it doesn't perform the approval process if the merkle tree to be approved is the same as the one in the blockchain.

### CODE REVIEW

File: src/modules/collection/sagas.ts [hunk 1]
```diff
@@ -668,27 +668,29 @@ export function* collectionSaga(legacyBuilderClient: BuilderAPI, client: Builder
...
 
       if (cheque.qty < amountOfItemsToPublish) {
         throw Error('Invalid qty of items to approve in the cheque')
       }
 
       // Open the ApprovalFlowModal with the items to be approved
       // 4. Make the transaction to the contract (update of the merkle tree root with the signature and its parameters)
-      if (itemsToApprove.length > 0) {
+      console.log('Root from the server: ', root)
+      console.log('Root from the client: ', tree.merkleRoot)
```
Comment (cazala) [inline src/modules/collection/sagas.ts:692]: oops

Comment (LautaroPetaccio) [reply src/modules/collection/sagas.ts]: Removed the console logs!

File: src/modules/collection/sagas.ts [hunk 2]
```diff
@@ -660,27 +660,30 @@ export function* collectionSaga(legacyBuilderClient: BuilderAPI, client: Builder
...
       const tree = generateTree(Object.values(contentHashes))
-      const amountOfItemsToPublish = itemsToApprove.filter(item => !item.isApproved).length
+      const amountOfItemsToPublish = itemsWithPendingCurations.filter(item => !item.isApproved).length
 
       if (cheque.qty < amountOfItemsToPublish) {
         throw Error('Invalid qty of items to approve in the cheque')
       }
 
+      console.log('Root value', root)
+      console.log('Tree root', tree.merkleRoot)
```
Comment (nicosantangelo) [inline src/modules/collection/sagas.ts:682]: 👀

Comment (LautaroPetaccio) [reply src/modules/collection/sagas.ts:682]: I removed the logs just to put them again in another commit LOL

**Review by nicosantangelo — APPROVED** [review]
LGTM

**Pull request merged**